axelor-account/src/main/java/com/axelor/apps/account/web/MoveLineQueryController.java

```diff
@@ -0,0 +1,135 @@
+      response.setValue("$selectedCredit", selectedCreditTotal);
+      response.setValue("$selectedBalance", selectedBalanceTotal);
+
+      String balanceTitle;
+      if (selectedBalanceTotal.signum() > 0) {
+        balanceTitle = I18n.get("Debit balance");
+      } else if (selectedBalanceTotal.signum() < 0) {
+        balanceTitle = I18n.get("Credit balance");
+      } else {
+        balanceTitle = I18n.get("Balance");
+      }
+      response.setAttr("$selectedBalance", "title", balanceTitle);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void massReconcile(ActionRequest request, ActionResponse response) {
+    try {
+      MoveLineQuery moveLineQuery = request.getContext().asType(MoveLineQuery.class);
+      moveLineQuery = Beans.get(MoveLineQueryRepository.class).find(moveLineQuery.getId());
+      List<MoveLine> moveLineList = new ArrayList<>();
+      if (!ObjectUtils.isEmpty(moveLineQuery.getMoveLineQueryLineList())) {
+        List<MoveLine> moveLineSelectedList =
+            moveLineQuery.getMoveLineQueryLineList().stream()
+                .filter(l -> l.getIsSelected())
+                .map(l -> l.getMoveLine())
+                .collect(Collectors.toList());
+        for (MoveLine moveLine : moveLineSelectedList) {
+          if (Beans.get(MoveLineControlService.class).canReconcile(moveLine)) {
+            moveLineList.add(moveLine);
+          }
+        }
+      }
+      if (!moveLineList.isEmpty()) {
+        Beans.get(MoveLineService.class).reconcileMoveLinesWithCacheManagement(moveLineList);
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void massUnreconcile(ActionRequest request, ActionResponse response) {
+    try {
+      MoveLineQuery moveLineQuery = request.getContext().asType(MoveLineQuery.class);
+      moveLineQuery = Beans.get(MoveLineQueryRepository.class).find(moveLineQuery.getId());
+      List<Reconcile> reconcileList = new ArrayList<>();
+      if (!ObjectUtils.isEmpty(moveLineQuery.getMoveLineQueryLineList())) {
+        List<MoveLine> moveLineSelectedList =
+            moveLineQuery.getMoveLineQueryLineList().stream()
+                .filter(l -> l.getIsSelected())
+                .map(l -> l.getMoveLine())
+                .collect(Collectors.toList());
+        for (MoveLine moveLine : moveLineSelectedList) {
+          for (Reconcile reconcile : moveLine.getDebitReconcileList()) {
+            if (!reconcileList.contains(reconcile)) {
+              reconcileList.add(reconcile);
+            }
+          }
+
+          for (Reconcile reconcile : moveLine.getCreditReconcileList()) {
+            if (!reconcileList.contains(reconcile)) {
+              reconcileList.add(reconcile);
+            }
+          }
+        }
+      }
+      if (!reconcileList.isEmpty()) {
+        Beans.get(MoveLineQueryService.class).ureconcileMoveLinesWithCacheManagement(reconcileList);
```

Comment:
Add a response.setReload(true) because we don't see any change otherwise. 
If you have time, improve the method MoveLineQueryService.ureconcileMoveLinesWithCacheManagement to return the number of reconciliations and display a message to inform the user about the number of reconciliations done.

Reply:
done without the message part